src/transformers/models/bridgetower/modeling_bridgetower.py

```diff
@@ -0,0 +1,1208 @@
+            "mean_pooling",
+            "lstm_uni",
+        ]:
+            self.LayerNorm = nn.LayerNorm(self.hidden_size * 2)
+
+        if self.downstream_fusion_method == "elmo":
+            self.learned_layer_weights = nn.Parameter(
+                torch.zeros(
+                    self.downstream_fusion_layers,
+                )
+            )
+        elif self.downstream_fusion_method == "elmo_tensor":
+            self.learned_layer_weights = nn.Parameter(torch.zeros(self.downstream_fusion_layers, self.hidden_size * 2))
+        elif self.downstream_fusion_method in ["gated", "gated_meter"]:
+            self.gate = nn.Linear(self.hidden_size * 2, self.hidden_size * 2)
+        elif self.downstream_fusion_method in ["lstm_uni", "lstm_bi"]:
+            if "uni" in self.downstream_fusion_method:
+                bidirectional = False
+            elif "bi" in self.downstream_fusion_method:
+                bidirectional = True
+            self.lstmpooler = nn.LSTM(
+                self.hidden_size * 2, self.hidden_size * 2, batch_first=True, bidirectional=bidirectional
+            )
+            self.lstmpooler_dropout = nn.Dropout(config.drop_rate)
+
+    def forward(self, feats):
+        if self.downstream_fusion_method == "concat":
+            return torch.cat(feats, dim=-1)
+        else:
+            feats = torch.stack(feats, dim=0)
+
+        if self.downstream_fusion_method == "ensemble":
+            return feats
+
+        if self.downstream_fusion_method in ["elmo", "elmo_tensor"]:
+            layer_weights = F.softmax(self.learned_layer_weights, dim=0)
+            if self.downstream_fusion_method == "elmo":
+                feats = torch.sum(layer_weights.view(-1, 1, 1) * feats, dim=0)
+            else:
+                feats = torch.sum(layer_weights.view(-1, 1, self.hidden_size * 2) * feats, dim=0)
+        elif self.downstream_fusion_method in ["gated", "gated_meter"]:
+            if self.downstream_fusion_method == "gated":
+                layer_gate = F.softmax(self.gate(feats), dim=0)
+                feats = torch.sum(layer_gate * feats, dim=0)
+            else:
+                layer_gate = F.softmax(self.gate(feats[:-1]), dim=0)
+                feats = feats[-1] + torch.sum(layer_gate * feats[:-1], dim=0)
+        elif self.downstream_fusion_method == "max_pooling":
+            feats = torch.max(feats, dim=0)[0]
+        elif self.downstream_fusion_method == "mean_pooling":
+            feats = torch.mean(feats, dim=0)
+        elif self.downstream_fusion_method == "max_mean_pooling":
+            max_feats = torch.max(feats, dim=0)[0]
+            mean_feats = torch.mean(feats, dim=0)
+            feats = torch.cat([max_feats, mean_feats], dim=-1)
+        elif self.downstream_fusion_method in ["lstm_uni", "lstm_bi"]:
+            feats = self.lstmpooler_dropout(self.lstmpooler(feats)[0][-1])
+
+        return self.LayerNorm(feats)
+
+
+class BridgeTowerPooler(nn.Module):
```

Comment:
if you intiliaze using the config, this class can be copied from BERT: https://github.com/huggingface/transformers/blob/1543cee7c8c95ef47f832b1f37625ba2923c4994/src/transformers/models/bert/modeling_bert.py#L652

Reply:
Handled in the updated commits.